docs/rfc/rfc-017-abci++-vote-extension-propag.md

```diff
@@ -346,21 +347,21 @@ These are the solutions proposed in discussions leading up to this RFC.
     Let us consider the following scenario:
 
     - all validators in *valset<sub>h</sub>* send out precommit messages, with vote extensions,
-      for height *h*, round 1, roughly at the same time,
+      for height *h*, round 0, roughly at the same time,
     - all those precommit messages contain non-`nil` precommit votes, which vote for block *b*
     - all validators in *valset<sub>h</sub>* then wait until they gather enough precommit
-      messages for height *h*, round 1, in order to decide in height *h*,
-    - all precommit messages sent in height *h*, round 1, and all messages sent in
-      height *h*, round *r > 1* get delayed indefinitely,
+      messages for height *h*, round 0, in order to decide in height *h*,
+    - all precommit messages sent in height *h*, round 0, and all messages sent in
+      height *h*, round *r > 0* get delayed indefinitely,
     - an intermediate (malicious) full node *m* manages to receive block *b*, and gather more than
-      *2n<sub>h</sub>/3* precommit messages for height *h*, round 1,
+      *2n<sub>h</sub>/3* precommit messages for height *h*, round 0,
     - one way or another, the solution should have a mechanism for a full node to tell another full
```

Comment:
I'm not exactly sure we need a way for a full node to 'tell' another full node it is late. We just need the full node to be able to determine it is late from the information on the network. It should be able to tell it is late by just examining the messages on the network including the blocks/prevotes/and precommits.

Reply:
Yes, that is another way of _modeling_ it. Rather than _m_ telling _n_ it is late, _m_ would send the commit and block for height 0 to show it is in height 1, then all full nodes _realize_ they are late, take the commit and block to decide and the extensions are lost.
If you don't mind, I'd like to keep the "catch-up message" way of explaining it, as I find it easier to explain like this. Let me know if you see a way to add one or two sentences saying "this problem can also be explained without the need of a catch-up message".

Reply:
I re-read this part of the text and decided to follow your comment and _model_ it both ways (_m_ sending the "catch-up message", and other full nodes concluding they are late by looking at messages from _m_). You may want to take a look at the new text